### DISCUSSION TITLE
Chain of commands before an assertion

### DISCUSSION THREAD

**Discussion by NormalCrazy000** [cypress-io/cypress#30525] [Questions and Help] [answered]
I was asking if for example :
```ssh
cy.get('#idDiv').find('p').should('have.lenght',1)
```
in case of failure of 'should' the whole chain of commands would be retried or only the command near the assertion, in this case 'find'

**Comment by portyu9** [comment, marked as answer]
Hi @NormalCrazy000 

The whole linked **query chain** is retried.

For:

```js
cy.get('#idDiv')
  .find('p')
  .should('have.length', 1)
```

if the assertion fails, Cypress starts again from `cy.get('#idDiv')`, then reruns `.find('p')`, then retries the assertion until it passes or the timeout is reached.

This applies because `get`, `find`, and `should` are linked queries/assertions. Non-query commands such as `.click()` are different: they execute once and are not rerun as part of later retries.

Also note the typo in the original example: it is `have.length`, not `have.lenght`.

Please mark this answer as accepted if it helped, thank you.
